src/main/java/com/lebediev/movieland/service/conversion/CachedCurrency.java

```diff
@@ -0,0 +1,93 @@
+package com.lebediev.movieland.service.conversion;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.security.Security;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CachedCurrency {
+    private final static Logger log = LoggerFactory.getLogger(CachedCurrency.class);
+    private volatile List<CurrencyEntity> currencyList;
+    private final static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    @Value("${currency.url}")
+    private String url; //= "https://bank.gov.ua/NBUStatService/v1/statdirectory/exchange?json";
+
+    public CurrencyEntity getCurrencyEntity(Currency currency){
+        log.info("Start getting currency from cache");
+        long startTime = System.currentTimeMillis();
+        List<CurrencyEntity> currencyEntityListCached = new ArrayList<>(currencyList);
+        for(CurrencyEntity currencyEntity : currencyEntityListCached){
+            if(currencyEntity.getCurrency() == currency){
+                log.info("Finish getting currency from cache. It took {} ms", System.currentTimeMillis() - startTime);
+                return currencyEntity;
+            }
+        }
+        throw new RuntimeException("Could not get cached currency " + currency);
+    }
+
+
+    public static List<CurrencyEntity> getCurrenciesFromUrl(String url){
```

Comment:
why public?

Reply:
thank you, changed. I had plans to test it somehow )